_rules/bypass-blocks-cf77f2.md

```diff
@@ -0,0 +1,412 @@
+- This rule assumes that the mean to bypass blocks is included in the content of the [HTML web page][]. For example, server side scripting can provide a functionality similar to [Block of content is expandable and collapsible][] by serving a modified version of the page; in which case this rule would fail but [Success Criterion 2.4.1: Bypass blocks][sc241] could nonetheless be satisfied.
+- This rule assumes that `frame` and `frameset` elements are not used, given that they are deprecated in HTML5. They can be used to organize content as per [H70: Using frame elements to group blocks of repeated material](https://www.w3.org/WAI/WCAG21/Techniques/html/H70) and [H64: Using the title attribute of the frame and iframe elements](https://www.w3.org/WAI/WCAG21/Techniques/html/H64), in which case this rule would fail but [Success Criterion 2.4.1: Bypass blocks][sc241] could nonetheless be satisfied.
+
+## Accessibility Support
+
+Techniques and solutions that identify blocks of content are sufficient ways of passing [Success Criterion 2.4.1: Bypass blocks][sc241]. They are, however, only beneficial for users who have ways of navigating with this information. For example, adding headings to a document will only help users who can "jump" from heading to heading (such a possibility can be provided by browsers, browsers plugins, screen readers, or other assistive technologies). Techniques and solutions based on links will benefit all users (for example, keyboard users with no other assistive technology) and are therefore recommended.
+
+## Background
+
+- [Understanding Success Criterion 2.4.1: Bypass Blocks][usc241]
+- Creating links to skip blocks of content:
+  - [G1: Adding a link at the top of each page that goes directly to the main content area](https://www.w3.org/WAI/WCAG21/Techniques/general/G1)
+  - [G123: Adding a link at the beginning of a block of repeated content to go to the end of the block](https://www.w3.org/WAI/WCAG21/Techniques/general/G123)
+  - [G124: Adding links at the top of the page to each area of the content](https://www.w3.org/WAI/WCAG21/Techniques/general/G124)
+- Grouping and identifying blocks of content:
+  - [ARIA11: Using ARIA landmarks to identify regions of a page](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA11)
+  - [H69: Providing heading elements at the beginning of each section of content](https://www.w3.org/WAI/WCAG21/Techniques/html/H69)
+  - [SCR28: Using an expandable and collapsible menu to bypass block of content](https://www.w3.org/WAI/WCAG21/Techniques/client-side-script/SCR28)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This [HTML web page][] is passing rule [Document Has no Repeated Content Before Main Content][].
```

Comment:
More descriptive text for example is recommended.

Consider this comment for other similar examples.

Reply:
For the composite rule, the Expectation is to pass one of the input rule. This example fulfils the expectation because it passes that rule. The reasons why it passes that rule are irrelevant for the composite rule and only matter for the atomic one.

On a more global level, I feel that keeping this kind of descriptions for composite rules, especially for examples that are taken directly from the atomic ones, is a good way to remind and enforce that the composite rule has no access to the input aspect. The only think that the composite rule knows is the output of its atomic rules, and it must make a decision based on these.